File: app/src/main/java/site/doramusic/app/http/ApiResult.java
package site.doramusic.app.http;

import dora.cache.data.adapter.Result;

public class ApiResult<T> implements Result<T> {

	private String errorCode;

	private String errorDetail;

	private T data;

	private long timestamp = System.currentTimeMillis();


	public String getErrorCode() {
		return errorCode;
	}

	public void setErrorCode(String errorCode) {
		this.errorCode = errorCode;
	}

	public long getTimestamp() {
		return timestamp;
	}

	public String getErrorDetail() {
		return errorDetail;
	}

	public void setErrorDetail(String errorDetail) {
		this.errorDetail = errorDetail;
	}

	public T getData() {
		return data;
	}

	public void setData(T data) {
		this.data = data;
	}

	@Override
	public T getRealModel() {
		return data;
	}
}

File: app/src/main/java/site/doramusic/app/http/ApiResult.kt
package site.doramusic.app.http

import dora.cache.data.adapter.Result

/**
 * 前后端交互数据标准。
 */
class ApiResult<T> : Result<T> {

    var code: String? = null
    var msg: String? = null
    var data: T? = null

    override fun getRealModel(): T? {
        return data
    }
}